Add Command sonic-clear logging

matiAlfaro opened the pull request:
<!--
    Please make sure you've read and understood our contributing guidelines:
    https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

    ** Make sure all your commits include a signature generated with `git commit -s` **

    If this is a bug fix, make sure your description includes "closes #xxxx",
    "fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
    issue when the PR is merged.

    If you are adding/modifying/removing any command or utility script, please also
    make sure to add/modify/remove any unit tests from the tests
    directory as appropriate.

    If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
    subcommand, or you are adding a new subcommand, please make sure you also
    update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
    your changes.

    Please provide the following information:
-->

#### What I did
Added command 
sonic-clear logging         - delete syslog & syslog.1
sonic-clear logging -a/--all     - delete syslog* files
#### How I did it
Equivalent to "show logging" command, working on files at /var/log or at /var/log.tmpfs
delete relevant syslog* files
#### How to verify it
1. clear logging when no syslog files exist
2. clear logging when only syslog file exist
3. clear logging when syslog & syslog.1 files exits
4. clear logging when syslog, syslog.1 and syslog2.gz files exits

Repeat 1-4 with
Clear logging --all

verify also that show logging behave correctly after clear
#### Previous command output (if the output of a command-line utility has changed)
NR
#### New command output (if the output of a command-line utility has changed)
none

Commit:
Add Command sonic-clear logging

Signed-off-by: mati <mati@marvell.com>

shiraez commented:
Looks good,
But it can be problematic if this PR merges:
https://github.com/sonic-net/sonic-utilities/pull/3132

matiAlfaro commented:
I think that one is already part of master, I'll have another look.
Was also thinking to clear the files instead of deleting them

VladimirKuk commented:
1. There are /var/log/syslog.X directories (where X is running number) which also contain some older log files.
2. Also, maybe we should add age parameter to delete "older than" files instead of deleting all of them.

Commit:
fix to delete syslog.1 when needed

Signed-off-by: mati <mati@marvell.com>

Commit:
add option --all to delete also store files

Signed-off-by: mati <mati@marvell.com>

shiraez commented:
BTW it looks like they added a test to the show logging

Commit:
type error fix

Signed-off-by: mati <mati@marvell.com>

Commit:
call run_command once

Signed-off-by: mati <mati@marvell.com>

Commit:
fix to work in all cases

Signed-off-by: mati <mati@marvell.com>

Commit:
type error fix

Signed-off-by: mati <mati@marvell.com>

amazor commented:
What Python version is used in SONiC?

amazor commented on clear/main.py:
```diff
@@ -550,6 +551,25 @@ def route(prefix, vrf, namespace):
+        files_to_delete += glob.glob("{}/syslog.1".format(log_path))
+
+    for f in files_to_delete:
+        cmd = ['sudo', 'rm' '-f',f]
```
Rather than deleting files using the shell, would it be better to use the built-in python `os.remove(f)`? There might be permission issues, but it is better practice rather than calling the shell.

matiAlfaro replied:
doesn't work due to privileges, couldn't find a way to run os.remove with sudo/root privileges.

amazor commented on clear/main.py:
```diff
@@ -550,6 +551,25 @@ def route(prefix, vrf, namespace):
+        log_path = "/var/log"
+    
+    if all:
+        files_to_delete = glob.glob("{}/syslog*".format(log_path))
```
Best practice is to use the updated (and cleaner) python format string:
`files_to_delete = glob.glob(f"{log_path}/syslog*")`

Commit:
fix type error

Signed-off-by: mati <mati@marvell.com>

amazor commented on clear/main.py:
```diff
@@ -550,6 +551,25 @@ def route(prefix, vrf, namespace):
+    if all:
+        files_to_delete = glob.glob("{}/syslog*".format(log_path))
+    else:
+        files_to_delete = glob.glob("{}/syslog".format(log_path))
```
No need for glob in this line, since there is no wildcard character

amazor commented on clear/main.py:
```diff
@@ -550,6 +551,25 @@ def route(prefix, vrf, namespace):
+        files_to_delete = glob.glob("{}/syslog*".format(log_path))
+    else:
+        files_to_delete = glob.glob("{}/syslog".format(log_path))
+    if os.path.isfile("{}/syslog.1".format(log_path))
```
Why check for `syslog.1` specifically if not clearing `--all` logs?
I would remove this line. If no `--all` option is requested, then only remove the current logs. Otherwise remove all backup logs as well, including `syslog.1`

matiAlfaro replied:
this is copying what is being done in the show logging

shiraez replied:
@matiAlfaro is right, if he doesn't delete it, the show logging command will take the syslog.1

matiAlfaro commented:
> What Python version is used in SONiC?

Not sure if correct for all platform/builds, but VS (master) runs 3.11.2
RD-AC5X (master) runs 3.9.2

Commit:
review fixes

Fixed Problems/New Changes:
change format syntax to new form
use glob only where needed

Should be Tested:

~

Signed-off-by: mati <mati@marvell.com>

Commit:
type error fix

Signed-off-by: mati <mati@marvell.com>

amazor reviewed (approved):
Looks good to me

matiAlfaro commented:
> BTW it looks like they added a test to the show logging

don't know if needed, don't think it's worth the effort, but if mandatory ...
